internal/workerpool/pool.go

```diff
@@ -22,27 +22,30 @@ import (
 	"context"
 )
 
-// Run runs a worker pool with no more than limit goroutines. It gets tasks
-// from the nextTask func. If nextTask returns nil, Run will no longer ask for
-// more tasks, and will return after waiting for the running goroutines to
-// finish. The doWork func processes a single task. The provided context can
-// be used to cancel everything and exit the loop.
-func Run(ctx context.Context, limit int, nextTask func(context.Context) interface{}, doWork func(ctx context.Context, task interface{})) {
+type Task func(context.Context)
```

Comment:
Maybe it doesn't need a context, because it's created inside nextTask?

Reply:
Technically I think you're right, but I want to make it clear that these tasks are expected to deal gracefully with cancellation.